flamingpy/codes/surface_code.py

```diff
@@ -88,18 +88,27 @@ def str_to_bound(bound_name):
 
         'open_primal': [primal, dual, dual]
         'open_dual': [primal, dual, primal]
+        'primal': [primal, primal, primal]
+        'dual': [dual, dual, dual]
+        'all_periodic': [periodic, periodic, periodic]
+        'periodic_primal': [periodic, periodic, primal]
+        'periodic_dual': [periodic, periodic, dual]
```

Comment:
In my opinion, the utility of string options is to give the user a convenient way to input common and meaningful choices of boundaries. Otherwise, they can input a list directly. Here, I would use the convention:

- If a single word, assume this choice of boundary in all three directions (i.e. replace `all_periodic` with `periodic`).
- If two words, assume common choice of boundaries for x, y (first word) and then boundary in z (second word). (i.e. everything else can stay).

For all other choices, a list suffices!

Reply:
Perfect! I'll apply this suggestions 🤓